### PR TITLE
[ADF-1711] The <adf-task-attachment-list component displays drag-and-drop area that is not working

### PR DESCRIPTION
**Pull request by madhukar23**
* Updated task attachment readme file for the user to know how to make drag and drop area working

* Changed activiti component to make uploading task attachment working by drag and drop in demo-shell

* Updated task attachment html file so that user can pass custom no-content template

* Added  a empty-list-directive in ng2-alfresco-tasklist

**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [x] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
The drag and drop is not working for task-attachment-list in demo-shell

**What is the new behaviour?**
made Drag and drop area working with the functionality to pass a custom no-content template

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: docs/task-attachment-list.component.md [hunk 1]
```diff
@@ -22,15 +23,36 @@ Displays attached documents on a specified task
 <adf-task-attachment-list 
     [taskId]="YOUR_TASK_ID" 
     (attachmentClick)="YOUR_HANDLER">
+            <div adf-empty-list>
+                {{Custom template for empty List}}
+            </div>
 </adf-task-attachment-list>
 ```
+### Drag and Drop Functionality
```
Comment (mauriziovitale) [inline docs/task-attachment-list.component.md]: can we change to 
How to add the Drag&Drop functionality

File: demo-shell-ng2/src/app/components/activiti/activiti.component.ts [hunk 2]
```diff
@@ -51,21 +53,27 @@ import {
 import { Subscription } from 'rxjs/Rx';
 import { /*CustomEditorComponent*/ CustomStencil01 } from './custom-editor/custom-editor.component';
 import { DemoFieldValidator } from './demo-field-validator';
-
+declare var require: any;
```
Comment (mauriziovitale) [inline demo-shell-ng2/src/app/components/activiti/activiti.component.ts]: why we need this declaration?

File: demo-shell-ng2/src/app/components/activiti/activiti.component.ts [hunk 3]
```diff
@@ -101,6 +109,7 @@ export class ActivitiComponent implements AfterViewInit, OnDestroy, OnInit {
     layoutType: string;
     currentTaskId: string;
     currentProcessInstanceId: string;
+    emptyListImageUrl: string = require('./../../../../../ng2-components/ng2-activiti-tasklist/src/assets/images/empty_doc_lib.svg');
```
Comment (mauriziovitale) [inline demo-shell-ng2/src/app/components/activiti/activiti.component.ts]: we don't need it

File: docs/task-attachment-list.component.md [hunk 4]
```diff
@@ -24,6 +25,30 @@ Displays attached documents on a specified task
...
+                <adf-empty-list>
+                    <div adf-empty-list-header>{{This List is empty}}</div>
+                    <div adf-empty-list-body>{{Drag and drop to upload}}</div>
+                    <div adf-empty-list-footer>
+                        <img [src]="Your custom image URL"></div> 
+                </adf-empty-list>
+            </div>
+    </adf-task-attachment-list>
+</adf-upload-drag-area>
+```
```
Comment (mauriziovitale) [inline docs/task-attachment-list.component.md]: can we attach an image of how the default template looks like and one related to the custom drag&drop template?

File: ng2-components/ng2-activiti-tasklist/src/components/task-attachment-list.component.html [hunk 5]
```diff
@@ -1,15 +1,20 @@
...
     <adf-empty-list *ngIf="isEmpty()">
-        <div adf-empty-list-header class="adf-empty-list-header"> {{'ADF_TASK_LIST.ATTACHMENT.EMPTY.HEADER' | translate}} </div>
-        <div adf-empty-list-body *ngIf="!isDisabled()">
-            <div fxHide.lt-md="true" class="adf-empty-list-drag_drop">{{'ADF_TASK_LIST.ATTACHMENT.EMPTY.DRAG-AND-DROP.TITLE' | translate}}</div>
-            <div fxHide.lt-md="true" class="adf-empty-list__any-files-here-to-add"> {{'ADF_TASK_LIST.ATTACHMENT.EMPTY.DRAG-AND-DROP.SUBTITLE' | translate}} </div>
-        </div>
-        <div adf-empty-list-footer *ngIf="!isDisabled()">
-            <img class="adf-empty-list__empty_doc_lib" [src]="emptyListImageUrl">
-        </div>
+            <div adf-empty-list-header class="adf-empty-list-header" *ngIf="myCustomEmptyList.children.length == 0 ">
```
Comment (mauriziovitale) [inline ng2-components/ng2-activiti-tasklist/src/components/task-attachment-list.component.html]: create a method called IsCustomTempateDefined

File: ng2-components/ng2-activiti-tasklist/src/components/task-attachment-list.component.html [hunk 6]
```diff
@@ -1,15 +1,20 @@
...
+                    {{'ADF_TASK_LIST.ATTACHMENT.EMPTY.HEADER' | translate}} 
+            </div>
+            <div #myCustomEmptyList>
```
Comment (mauriziovitale) [inline ng2-components/ng2-activiti-tasklist/src/components/task-attachment-list.component.html]: call the referemce customEmptyListTemplate

File: demo-shell-ng2/src/app/components/activiti/activiti.component.ts (hunk 3 above)
Comment (madhukar23) [reply demo-shell-ng2/src/app/components/activiti/activiti.component.ts]: Don't we need to show the image also with our custom template?

**Pull request closed without merging**